packages/core/src/deployment.ts

```diff
@@ -73,11 +74,15 @@ export async function waitAndValidateDeployment(provider: EthereumProvider, depl
   }
 
   debug('succeeded verifying deployment', txHash);
-  if (await hasCode(provider, address)) {
-    return;
+  while (!(await hasCode(provider, address))) {
+    const startTime = Date.now();
```

Comment:
`startTime` should be outside the loop!

```suggestion
  const startTime = Date.now();
  while (!(await hasCode(provider, address))) {
```